nav2_bt_navigator/behavior_trees/nav2_tree_nodes.xml

```diff
@@ -0,0 +1,66 @@
+<root>
```

Comment:
Can you add some comment about checking out the readme for information on use of this? 

Do you think this belongs here or in the `/tools` directory? I'm not 100% sure.

Reply:
> Can you add some comment about checking out the readme for information on use of this?

I'll add that. Do you think it should also be specified [here](https://github.com/ros-planning/navigation2/blob/master/nav2_bt_navigator/README.md)? or let it be because it will eventually be moved to the website.

> Do you think this belongs here or in the /tools directory? I'm not 100% sure.

I think I have placed it in the right location. It's not exactly a tool but a list that summarizes all the BT nodes used in navigation2 along with their port descriptions, which will provide node metadata to Groot. This should be kept with the behavior trees.

Reply:
I think just having it in the behavior tree pkg is good enough since that's the package someone would look at if they're trying to modify the tree. The BT navigator is just running things at execution time. You're right long term this should be in the website, so I'm not too picky about the location as long as the content is there as a reminder to add to the website. 

Since this isn't actually a behavior tree that can be executed, its a list of nodes for groot, this seems to me to be more of a tool analog to the bt2img script thats in `tools/`.  I'd agree it should be here for sure if it was required for run-time execution but seems more like an editor thing. Since its not for the navigator, maybe even putting it in `nav2_behavior_trees` instead seems to be the best location for the same reason as the readme stuff in the paragraph above. That's the package for editing and adding new content / capabilities. BT navigator just runs it. I think just adding this to the root of that package would good, just make sure to run install rules over it. 

I'd be fine with it here if we had a way to clearly separate the nodes file from the trees.

Reply:
The main reason I kept it under `behavior_trees` folder is to keep it in the same directory with the behavior trees.
>  nav2_behavior_trees instead seems to be the best location for the same reason as the readme stuff in the paragraph above. That's the package for editing and adding new content / capabilities.

I think it is a good idea. It will be easier if all of the base node descriptions are kept in the same place. It will remove the hassle to edit and create new nodes and update their descriptions if all of them are in one package.

> just make sure to run install rules over it.

If I understand that correctly, you mean to say Groot installation procedure? If yes, it should be added to the README.md of instruction to use with Groot. I have provided the link to the main repo though, I'll mention it here too.

Reply:
No, just run the cmake install rules so that this xml file in the root of the package is installed.

Reply:
Sorry but I fail to understand why this is required. Can you help me out?

This file is used with Groot and is not required anywhere else.

Reply:
If its in the source code but not `install()`-ed in the CMakeLists, its not installed in binary distributions. Eg if someone wants to view a BT on their computer with an `apt-get install` version of navigation, without the install rule, they will not get it because it won't be copied into the install space.

Reply:
Got it! Thanks!